test/play-item.test.js

```diff
@@ -80,7 +81,70 @@ QUnit.test(
 
     assert.deepEqual(
       cues,
-      [{ time: 0, type: 'foo' }, { time: 1, type: 'bar' }],
+      [{startTime: 0, endTime: 0.01667, type: 'foo' },
+      {startTime: 1, endTime: 1.01667, type: 'bar' }],
+      'cues are what we expected'
+    );
+    window.VTTCue = oldVttCue;
+  }
+);
+
+QUnit.test(
+  'Backwards compatibility test for setting sources, poster, tracks and cue points',
+  function(assert) {
+    let oldVttCue = window.VTTCue;
+    let player = playerProxyMaker();
+    let setSrc;
+    let setPoster;
+    let setTracks = [];
+    let cues = [];
+
+    window.VTTCue = (time, timeEnd, type) => ({time, type});
+
+    player.src = function(src) {
+      setSrc = src;
+    };
+
+    player.poster = function(poster) {
+      setPoster = poster;
+    };
+
+    player.addRemoteTextTrack = function(tt) {
+      setTracks.push(tt);
+      return {
+        track: {
+          addCue(cue) {
+            cues.push(cue);
+          }
+        }
+      };
+    };
+
+    playItem(player, null, {
+      sources: [1, 2, 3],
+      textTracks: [4, 5, 6],
+      poster: 'http://example.com/poster.png',
+      cuePoints: [{time: 0, type: 'foo' },
+      {time: 1, type: 'bar' }]
+    });
+
+    assert.deepEqual(setSrc, [1, 2, 3], 'sources are what we expected');
+    assert.deepEqual(
+      setTracks.sort(),
+      [4, 5, 6, { kind: 'metadata' }].sort(),
+      'tracks are what we expected'
+    );
+
+    assert.equal(
+      setPoster,
+      'http://example.com/poster.png',
+      'poster is what we expected'
+    );
+
+    assert.deepEqual(
+      cues,
+      [{time: 0, type: 'foo' },
+      {time: 1, type: 'bar' }],
```

Comment:
I don't think this actually guarantees backwards compatibility. What I imagined you'd be testing is that cues like this `{ time: 14, type: 'whatever' }` would get translated into VTTCues like this `new VTTCue(14, 14.0001, 'whatever')`. It doesn't seem like that's occurring based on this assertion.